Description: Fix memory leak in date if the gc is enabled
Origin: http://svn.php.net/viewvc?view=revision&revision=298222
Last-Update: 2010-05-01

Index: php/ext/date/php_date.c
===================================================================
--- php.orig/ext/date/php_date.c
+++ php/ext/date/php_date.c
@@ -2081,7 +2081,7 @@ static HashTable *date_object_get_proper
 
 	props = dateobj->std.properties;
 
-	if (!dateobj->time) {
+	if (!dateobj->time || GC_G(gc_active)) {
 		return props;
 	}
 
@@ -2224,7 +2224,7 @@ static HashTable *date_object_get_proper
 
 	props = intervalobj->std.properties;
 
-	if (!intervalobj->initialized) {
+	if (!intervalobj->initialized || GC_G(gc_active)) {
 		return props;
 	}
 
Index: php/ext/date/tests/bug49700.phpt
===================================================================
--- /dev/null
+++ php/ext/date/tests/bug49700.phpt
@@ -0,0 +1,15 @@
+--TEST--
+Bug #49700 (memory leaks in php_date.c if garbage collector is enabled)
+--INI--
+date.timezone=GMT
+--FILE--
+<?php
+gc_enable();
+$objs = array();
+$objs[1] = new DateTime();
+gc_collect_cycles();
+unset($objs);
+echo "OK\n";
+?>
+--EXPECT--
+OK
